packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -1064,6 +1101,7 @@ export class IntervalCollection<TInterval extends ISerializableInterval>
             // This is an ack from the server. Remove the pending change.
             this.removePendingChange(serializedInterval);
             const id: string = serializedInterval.properties[reservedIntervalIdKey];
+            // Could store the interval in the localOpMetadata to avoid the getIntervalById call
```

Comment:
don't think this works:

1. Locally, it's possible there's a local delete of this interval before the change gets acked. Ofc we could deal with this differently e.g. by tracking that pending delete, but it's not feasible for the change's local op metadata to know about this case
2. local op metadata wouldn't be available remotely (unless you just meant strictly as a perf optimization for the local case)

Reply:
1. I was thinking the metadata would hold the actual interval. It should be pretty easy to handle delete. 
2. isn't an issue as this code only runs on the ack of a local operation.

Reply:
Oh, didn't realize you meant this comment for local case only (though it is in the block, my bad... still getting used to github diff UI).

Yeah, I also observed a few places in interval collection where local op metadata could be used more efficiently. E.g. I think the current implementation of pending changes is a bit difficult to follow (and inefficient, since it uses an array as a queue IIRC) and would be simpler if it took the map approach of tracking pending ops by some id associated to local op metadata.

If we take the approach you mention here, we'd need to also revisit how `change` works on the local collection since it really removes the existing one and adds the modified one to the RB tree.